File: Three Consecutive Odds.py
# https://leetcode.com/problems/three-consecutive-odds/?envType=daily-question&envId=2024-07-01
# Three Consecutive Odds

class Solution:
    def threeConsecutiveOdds(self, arr: List[int]) -> bool:

        odd = 0
        
        for num in arr:

            if num%2 == 1:
                odd += 1
            else:
                odd = 0

            if odd == 3:
                return True

        return False

File: Three Consecutive Odds.js
// https://leetcode.com/problems/three-consecutive-odds/?envType=daily-question&envId=2025-05-11
// Three Consecutive Odds

/**
 * @param {number[]} arr
 * @return {boolean}
 */
var threeConsecutiveOdds = function(arr) {

    let odd_count = 0;
    
    for(let num of arr) {
        if (num % 2 !== 0) odd_count++
        else odd_count = 0

        if (odd_count === 3) return true;
    };

    return false;

};